### ISSUE TITLE
[BUG] Metric definition is not detected

### ISSUE THREAD

**Issue by Rizhiy** [aws/sagemaker-python-sdk#3932]
**Describe the bug**
I'm trying to run a HyperParameter Tuning job, but it gives me an error:
```
An error occurred (ValidationException) when calling the CreateHyperParameterTuningJob operation: A metric is required for this hyperparameter tuning job objective. Provide a metric in the metric definitions.
```
I have defined relevant metrics in Estimator definition.

**To reproduce**
```python
container = sagemaker.image_uris.retrieve(
    "sklearn", boto3.Session().region_name, "1.2-1"
)
# container = sagemaker.image_uris.retrieve(
#     "xgboost", boto3.Session().region_name, "latest"
# )
xgb_estimator = Estimator(
    image_uri=container,
    source_dir="scripts",
    entry_point="train.py",
    instance_type="ml.m5.large",
    instance_count=1,
    output_path=f"s3://{bucket}/output",
    role=role,
    metric_definitions=[{"Name": "validation:f1", "Regex": "F1=(.*?)"}],
)

objective_metric_name = "validation:f1"

hyperparameter_ranges = {
    "n_estimators": IntegerParameter(5, 500, scaling_type="Logarithmic"),
    "learning_rate": ContinuousParameter(0.01, 1, scaling_type="Logarithmic"),
    "booster": CategoricalParameter(["gbtree", "gblinear", "dart"]),
}

tuner = HyperparameterTuner(
    xgb_estimator,
    objective_metric_name,
    objective_type="Maximize",
    hyperparameter_ranges=hyperparameter_ranges,
    max_jobs=16,
    max_parallel_jobs=4,
    strategy="Random",
)

tuner.fit({"data": TrainingInput(s3_data_path)})
```

**Expected behavior**
Start the tuning job

**System information**
A description of your system. Please provide:
- **SageMaker Python SDK version**: 2.146.0
- **Framework name (eg. PyTorch) or algorithm (eg. KMeans)**: SKLearn
- **Framework version**: 1.2-1
- **Python version**: 3.10.8
- **CPU or GPU**: CPU
- **Custom Docker image (Y/N)**: N

**Additional context**
None

**Comment by martinRenou** [comment]
Closing as answered. Please feel free to continue the discussion if needed.
